src/parseUnitNumber.ts

```diff
@@ -0,0 +1,129 @@
+export interface IParseUnitNumberResult {
+  civicAddress: string;
+  unitDesignator: string;
+  unitNumber: string;
+}
+
+// Mapping of full unit designator string to preferred abbreviation.
+const normalizedDesignators = {
+  '#': '',
+  apartment: 'apt',
+  building: 'bldg',
+  department: 'dept',
+  floor: 'fl',
+  room: 'rm',
+  suite: 'ste',
+  unit: 'unit'
```

Comment:
what about the french versions of these? Canada Post lists some of them here https://www.canadapost.ca/tools/pg/manual/PGaddress-e.asp?ecid=murl10006450#1442083